README.rst

```diff
@@ -74,11 +68,140 @@ To install in a virtual environment in your current project:
+
+Commands
+---------
+
+Commands always start with a fixed header. The first entry is always the command number itself encoded in a single byte. The number of subsequent entries in the header will vary by command. The entire header must be sent as a unit so set the characteristic with the full header packet.
+
+Paths use `/` as a separator and full paths must start with `/`. Directory paths
+must end with `/` when provided as a full path.
+
+All numbers are unsigned.
+
+Status bytes are `0x81` for OK and `0x82` for error. Other values for error may be used for specific commands.
+
+`0x01` - Read a file
+++++++++++++++++++++
+
+Given a full path, returns the full contents of the file.
+
+The header is three fixed entries and a variable length path:
+
+* Command: Single byte. Always `0x01`.
+* Chunk size: 32-bit number encoding the amount of data that the client can handle in the first reply.
+* Path length: 16-bit number encoding the encoded length of the path string.
+* Path: UTF-8 encoded string that is *not* null terminated. (We send the length instead.)
+
+The server will respond with:
+* Command: Single byte. Always `0x01`.
+* Status: Single byte.
+* Total length: 32-bit number encoding the total file length.
+* Chunk length: 32-bit number encoding the length of the read data up to the chunk size provided in the header.
+* Chunk-length contents of the file starting from the current position.
+
+If the chunk length is smaller than the total length, then the client will request more data by sending:
+* Command: Single byte. Always `0x01`.
+* Status: Single byte. Always OK for now.
+* Chunk size: 32-bit number encoding the number of bytes to read. May be different than the original size. Does not need to be limited by the total size.
+
+The transaction is complete after the server has replied with all data. (No acknowledgement needed from the client.)
+
+`0x02` - Write a file
++++++++++++++++++++++
+
+Writes the content to the given full path. If the file exists, it will be overwritten. Content may be written as received so an interrupted transfer may lead to a truncated file.
+
+The header is three fixed entries and a variable length path:
+
+* Command: Single byte. Always `0x02`.
+* Total size: 32-bit number encoding the total length of the file contents.
+* Path length: 16-bit number encoding the encoded length of the path string.
+* Path: UTF-8 encoded string that is *not* null terminated. (We send the length instead.)
+
+The server will repeatedly respond until the total length has been transferred with:
+* Command: Single byte. Always `0x02`.
+* Status: Single byte. `0x81` if OK. `0x82` if any parent directory is missing or a file.
```

Comment:
"or a file"?  I don't follow.

Reply:
If a parent in the path is a file rather than a directory.